### DISCUSSION TITLE
Multicast Support

### DISCUSSION THREAD

**Discussion by B08Z** [alecbcs/hyprspace#5] [Ideas]
This may be a silly idea but would multicast support be a good addition. I can certainly come up with a few use cases for it so it would be a great addition.

Great work so far!

**Comment by alecbcs** [comment]
Haha thank you! I think multicast would be an interesting idea! I suppose the easiest way of implementing it would be to allow multiple peers to use a single IP address and then relay packets to all of the peers assigned to that address. Although this isn't true multicast at the base network level since you'd still have to duplicate the packets to multiple hosts, is this what you were thinking?

**Reply by B08Z** [reply]
Hey thanks for the reply. 
I like that idea but it would make it difficult if you wanted to send data to just one node. 

If you could have a group address so that you have your individual address but can join a group and within that everyone gets everything...

**Reply by alecbcs** [reply]
Sure! I think it could be possible to add multi-cast support by allowing you to define a IP address that maps anywhere from 1 to many peers.
